risc0/zkvm/platform/src/syscall.rs

```diff
@@ -218,8 +220,8 @@ macro_rules! impl_syscall {
                 ::core::arch::asm!(
                     "ecall",
                     in("t0") $crate::syscall::ecall::SOFTWARE,
-                    inout("a0") from_host => a0,
-                    inout("a1") from_host_words => a1,
+                    inlateout("a0") from_host => a0,
```

Comment:
me too!

Reply:
Going off of [the `asm!` RFC](https://rust-lang.github.io/rfcs/2873-inline-asm.html#late-output-operands), it seems to just be an optimization. It seems `inout` like before would have also been correct.

Reply:
That was my take as well. In any case, the compiler was emitting incorrect instructions (it was missing the load of the lower bits of the immediate value). An alternative fix could be to use `.s` files so that we have more precise control of the instructions. 

I'll try the alternative out because I don't trust this `inlateout` fix.

Reply:
For reference, could you please share the previous `inout` codegen and new `inlateout` version? It might be worth filing an issue

Reply:
Issue filed: https://github.com/rust-lang/rust/issues/128212